configserver/src/main/java/com/yahoo/vespa/config/server/session/PrepareParams.java

```diff
@@ -153,6 +154,12 @@ public Builder endpointCertificateMetadata(String serialized) {
         }
 
         public Builder dockerImageRepository(String dockerImageRepository) {
+            if (dockerImageRepository == null) return this;
```

Comment:
The `null` behaviors in this class are all over the place. `endpointCertificateMetadata(String)`, `containerEndpoints(String)` ignores it, as does `athenzDomain(AthenzDomain)`, while `athenzDomain(String)`, and `tlsSecretsKeyName(String)` (correctly, IMO) resets the state.

Reply:
Yes, good point, I'll try to clean it up